soundata/datasets/singapura.py

```diff
@@ -0,0 +1,379 @@
+        super().__init__(
+            clip_id,
+            data_home,
+            dataset_name=dataset_name,
+            index=index,
+            metadata=metadata,
+        )
+
+        self.audio_path = self.get_path("audio")
+        self.annotation_path = self.get_path("annotation")
+
+    @core.cached_property
+    def events(self) -> Optional[annotations.Events]:
+        """
+        The clip's event annotations
+
+        Returns:
+            * annotations.Events - sound events with start time, end time, label and confidence
+        """
+        return load_annotation(self.annotation_path)
+
+    @property
+    def audio(self):
+        """
+        The clip's audio
+
+        Returns:
+            * np.ndarray - audio signal
+        """
+        return load_audio(self.audio_path)
+
+    @property
+    def sensor_id(self) -> str:
+        """
+        The clip's sensor ID
+
+        Returns:
+            * str - sensor_id of the device used to record the data
+        """
+        return self._clip_metadata["sensor_id"]
+
+    @property
+    def town(self) -> str:
+        """
+        The clip's location
+
+        Returns:
+            * str - location of the sensor, one of {'East 1', 'East 2', 'West 1', 'West 2'}
+        """
+        return self._clip_metadata["sensor_id"]
+
+    @property
+    def timestamp(self) -> np.datetime64:
+        """
+        The clip's timestamp
+
+        Returns:
+            * np.datetime64 - timestamp of the clip
+        """
+
+        if self._clip_metadata["timezone"] == "SGT":
+            tz = "+08:00"
+        else:
+            # currently all of our data are in GMT+08:00
+            raise NotImplementedError
+
+        return np.datetime64(
+            f"{self._clip_metadata['year']}-{self._clip_metadata['month']:02d}-{self._clip_metadata['date']:02d}"
+            + f"T{self._clip_metadata['hour']:02d}:{self._clip_metadata['minute']:02d}:{self._clip_metadata['second']:02d}{tz}"
+        )
+
+    @property
+    def dotw(self) -> int:
+        """
+        The clip's day of the week
+
+        Returns:
+            * int - day of the week when the clip was recorded, starting from 0 for Sunday
+        """
+        return self._clip_metadata["day"]
+
+    def to_jams(self):
+        """
+        Jams: the track's data in jams format
+        """
+        return jams_utils.jams_converter(
+            audio_path=self.audio_path, events=self.events, metadata=self._clip_metadata
+        )
+
+
+@io.coerce_to_string_io
+def load_annotation(fhandle: TextIO) -> Optional[annotations.Events]:
+    """
+    Load an annotation file.
+
+    Args:
+        fhandle (str or file-like): path or file-like object pointing to an annotation file
+
+    Returns:
+        * annotations.Events - sound events with start time, end time, label and confidence
+    """
+
+    df = pd.read_csv(fhandle)
+
+    for _, dfa in df.groupby("annotator"):
```

Comment:
Does this dataset have multiple annotators?

Reply:
Yes, we have multiple annotators for some of the clips (not all). I was having issues with typing using `MultiAnnotator` so I changed everything to single-annotator. Let me give it another crack.